### PR TITLE
feat: change the AI command line argument to optional

### PR DESCRIPTION
**Pull request by jueli12**
## What type of PR is this?

/kind feature

## What this PR does / why we need it:

change the AI command line argument to optional

## Which issue(s) this PR fixes:

Fixes #551

### CODE REVIEW

File: pkg/core/handler/search/search.go [hunk 1]
```diff
@@ -69,7 +67,12 @@ func SearchForResource(searchMgr *search.SearchManager, aiMgr *ai.AIManager, sea
 		query := searchQuery
 
 		if searchPattern == storage.NLPatternType {
-			//logger.Info(searchQuery)
+			// logger.Info(searchQuery)
```
Comment (elliotxx) [inline pkg/core/handler/search/search.go:70]: Useless comments can be removed

File: pkg/core/manager/ai/manager.go [hunk 2]
```diff
@@ -25,7 +25,10 @@ type AIManager struct {
 
 // NewAIManager returns a new AIManager object
 func NewAIManager(c registry.ExtraConfig) (*AIManager, error) {
-	aiClient := ai.NewClient(c.Backend)
+	if c.AIAuthToken == "" {
+		return nil, nil
```
Comment (elliotxx) [inline pkg/core/manager/ai/manager.go:29]: If the token is empty, an error should be returned, otherwise the npe error is prone to occur

Comment (jueli12) [reply pkg/core/manager/ai/manager.go:29]: When Karpor starts, it creates a new AIManager. If return the error, the startup will fail. Is there any other way to do this?

Comment (elliotxx) [reply pkg/core/manager/ai/manager.go:29]: It should be up to the caller to decide which errors to ignore. The underlying method should ensure that the returned entity is not empty when there are no errors.

**Review by elliotxx — APPROVED** [review]
lgtm

**Pull request merged**